languages/python/tests/test_polar.py

```diff
@@ -556,8 +557,44 @@ def __eq__(self, other):
     assert query(Predicate("gt", [a2, a1]))
 
 
-def test_datetime(polar, query):
+def test_built_in_specializers(polar, query):
+    polar.load_str("foo(x: Boolean) if x = true;")
+    polar.load_str("foo(x: Integer) if x > 2;")
+    polar.register_constant("pi", pi)
+    polar.load_str("foo(x: Float) if x > pi;")
+    polar.load_str('foo(x: List) if x = ["foo", *_rest];')
+    polar.load_str('foo(x: Dictionary) if x.foo = "foo";')
+    polar.load_str('foo(x: String) if x = "foo";')
+
+    # Boolean
+    assert query("foo(true)")
+    assert not query("foo(false)")
+
+    # Integer
+    assert query("foo(3)")
+    assert not query("foo(2)")
+    assert not query("foo(-2)")
+
+    # Float
+    assert query("foo(4)")
```

Comment:
Why does this match the float rule instead of the integer one?

Reply:
It doesn't. Thanks.